assets/js/editor/Editor.tsx

```diff
@@ -313,7 +314,13 @@ export default function Editor({
         defaultPath="/job.js"
         loading={<div className="text-white">Loading...</div>}
         value={source || DEFAULT_TEXT}
-        options={options}
+        options={{
+          ...options,
+          readOnly: disabled,
+          readOnlyMessage: {
+            value: disabledMessage,
+          },
+        }}
```

Comment:
@josephjclark could I get your opinion on this? I'm concerned this might defeat the render/props checking causing re-renders?

Reply:
Yeah this is probably fine. You're right to ask and this pattern is usually a no-no with react components for that reason.

But often less so with options. I expect Monaco will deep diff the options before triggering a re-render, which case we're golden.

Reply:
Having said that! I also see that `options` is held in a `useState`, and it would likely be cleaner to drive the state options properly rather than overriding here